spec/services/determine_training_record_state_legacy_spec.rb

```diff
@@ -1137,6 +1137,30 @@
                          "not_yet_mentoring"
       end
 
+      context "and their mentee is leaving" do
```

Comment:
Its a pain because we have the logic duplicated just now, but it may be worth duplicating these specs here as well:

https://github.com/DFE-Digital/early-careers-framework/blob/4ceecee858ba566983ea3405ca160454d8874c2c/spec/support/shared_examples/optimised_training_record_state_query_support.rb#L16

Reply:
Yes good idea, I've added those